presto-spi/src/main/java/com/facebook/presto/spi/plan/AggregationNode.java

```diff
@@ -210,7 +210,12 @@ public PlanNode replaceChildren(List<PlanNode> newChildren)
 
     public boolean isStreamable()
```

Comment:
We need to check if any of the callsite also applies to the SegementedAggregation case.
For example, PushPartialAggregationThroughJoin::isSupportedAggregationNode returned false for streaming aggregation. There is also a usage in HashGenerationOptimizer.java. I haven't check the details about these rules yet. @mbasmanova, do you know what should be the expected value for segmented aggregation?

Reply:
1. isStreamable is used in two places out of all the optimizer rules, similarly we add isSegmentedAggregationEligible in those places as well
- HashGenerationOptimizer
For segmented aggregation, we shouldn't calculate hash for all the fields, but only for the fields that need to compute hash, this requires more changes and benchmark result to support such optimization, left a todo for future support 

- PushPartialAggregationThroughJoin
It only pushes Partial aggregation, for Partial aggregation, we currently haven't added segmented aggregation, so we should be safe there

2. Regarding the aggregation pushdown, there are two places
-  PushPartialAggregationThroughExchange
According to the comments, we would anyway drop the property, so for segmented aggregation, the original behavior is not affected
// preGroupedSymbols reflect properties of the input. Splitting the aggregation and pushing partial aggregation
// through the exchange may or may not preserve these properties. Hence, it is safest to drop preGroupedSymbols here.

-  PushPartialAggregationThroughJoin
Explained above